### ISSUE TITLE
[BUG] Cannot get API Manager config with user with Org Admin role

### ISSUE THREAD

**Issue by vuvdoan** [Axway-API-Management-Plus/apimanager-swagger-promote#116]
**Swagger-Promote Version**
1.6.0

**API-Manager and Service-Pack Version**
7.5.3 SP9

**Expected behavior**
use logged-in user with "Organization Admin role" to get   API-Manager version

**Actual behavior**
 successful logged in:
```930  [main] DEBUG org.apache.http.wire  - http-outgoing-0 >> "GET /api/portal/v1.3/currentuser HTTP/1.1[\r][\n]"
...
930  [main] DEBUG org.apache.http.wire  - http-outgoing-0 >> "Cookie: APIMANAGERSESSION=73148eee-acbb-45d3-8d51-88359b7499a9; APIMANAGERSTATIC=93c8e5d5-9ccf-4465-b12d-44d36acd7ef4; ROUTEID=.2[\r][\n]"
...
1068 [main] DEBUG org.apache.http.wire  - http-outgoing-0 << "Server: [\r][\n]"
1068 [main] DEBUG org.apache.http.wire  - http-outgoing-0 << "Expires: Mon, 22 Jul 2019 13:28:06 GMT[\r][\n]"
1068 [main] DEBUG org.apache.http.wire  - http-outgoing-0 << "CSRF-Token: 4EDF7D53CEA43FB7AA63900C27B543B69901880A6F7CDDC99E819838E3ED843D[\r][\n]"
1068 [main] DEBUG org.apache.http.wire  - http-outgoing-0 << "Content-Type: application/json[\r][\n]"
1068 [main] DEBUG org.apache.http.wire  - http-outgoing-0 << "X-CorrelationID: Id-e6b9355d97c17bb34118929e 0[\r][\n]"
1068 [main] DEBUG org.apache.http.wire  - http-outgoing-0 << "Set-Cookie: APIMANAGERSESSION=73148eee-acbb-45d3-8d51-88359b7499a9;Version=1;Comment="Session for API Management";Path=/api/portal/v1.3/;Secure;HttpOnly[\r][\n]"
1068 [main] DEBUG org.apache.http.wire  - http-outgoing-0 << "Set-Cookie: APIMANAGERSTATIC=93c8e5d5-9ccf-4465-b12d-44d36acd7ef4;Version=1;Comment="Session for API Management Static Files";Path=/;Secure;HttpOnly[\r][\n]"
1068 [main] DEBUG org.apache.http.wire  - http-outgoing-0 << "Keep-Alive: timeout=5, max=99[\r][\n]"
1068 [main] DEBUG org.apache.http.wire  - http-outgoing-0 << "Connection: Keep-Alive[\r][\n]"
1068 [main] DEBUG org.apache.http.wire  - http-outgoing-0 << "Transfer-Encoding: chunked[\r][\n]"
1068 [main] DEBUG org.apache.http.wire  - http-outgoing-0 << "[\r][\n]"
1068 [main] DEBUG org.apache.http.wire  - http-outgoing-0 << "1a3[\r][\n]"
1068 [main] DEBUG org.apache.http.wire  - http-outgoing-0 << "{"id":"bac824f4-ecc8-4d7e-997e-xxx","organizationId":"68af1395-f42f-48ed-9117-xxx","name":"xxx","description":null,"loginName":"xxx","email":"xxx","phone":null,"mobile":null,"role":"oadmin","image":null,"enabled":true,"createdOn":1556290887949,"state":"approved","type":"external","authAttrs":null,"dn":"cn=xxx"}[\r][\n]"
```

BUT failed to load the API Manger configuration information:
```
1869 [main] DEBUG org.apache.http.impl.execchain.MainClientExec  - Executing request GET /api/portal/v1.3/config HTTP/1.1
```
but get 
```
2026 [main] DEBUG org.apache.http.wire  - http-outgoing-1 << "HTTP/1.1 401 Unauthorized[\r][\n]"
```

**Test case sample**
Run the tool with any API configuration using an user with only Org Admin role.

**Comment by cwiechmann** [comment]
Hi @vuvdoan, 

This is a restriction of the API-Manager, that Org-Admins can’t read the actual configuration. Swagger-Promote tries to handle this as best as it can.

Do you have any specific issue, that Swagger-Promote doesn’t handle it properly?

**Comment by vuvdoan** [comment]
yeah.
 I did two curls with one user with Org-Admin role and one with API Manager Admin role. Although the curl with org-admin role doesn't have the same response, but it doesn't failed with a 401 error either:

curl with org-admin
```
{
  "registrationEnabled": false,
  "minimumPasswordLength": 6,
  "baseOAuth": false,
  "resetPasswordEnabled": false,
  "loginNameRegex": "[^;,\\\\/?#<>&;!]{1,}",
  "isTrial": false,
  "promoteApiViaPolicy": false,
  "systemOAuthScopesEnabled": true,
  "apiPortalName": "API Portal",
  "productVersion": "7.5.3 SP9",
  "delegateUserAdministration": true,
  "portalName": "Axway API Manager",
  "autoApproveApplications": false,
  "autoApproveUserRegistration": false,
  "delegateApplicationAdministration": true
}
```

curl with API Manager admin:
```
{
  "registrationEnabled": false,
  "minimumPasswordLength": 6,
  "baseOAuth": false,
  "resetPasswordEnabled": false,
  "loginNameRegex": "[^;,\\\\/?#<>&;!]{1,}",
  "emailBounceAddress": "xxx",
  "defaultTrialDuration": 30,
  "isTrial": false,
  "promoteApiViaPolicy": false,
  "apiPortalHostname": "",
  "systemOAuthScopesEnabled": true,
  "apiPortalName": "API Portal",
  "productVersion": "7.5.3 SP9",
  "sessionIdleTimeout": 3600000,
  "isApiPortalConfigured": false,
  "delegateUserAdministration": true,
  "portalName": "Axway API Manager",
  "autoApproveApplications": false,
  "autoApproveUserRegistration": false,
  "delegateApplicationAdministration": true,
  "apiDefaultVirtualHost": "xxx",
  "emailFrom": "xxx",
  "portalHostname": "xxx"
}
```

That leads to  a later problem with the initializing faultHandlerPolicies:
```
if(!APIManagerAdapter.getApiManagerVersion().startsWith("7.5")) {
				OutboundProfile.apimFaultHandlerPolicies = initPolicyies("faulthandler");
			}
```

APIManagerAdapter.getApiManagerVersion() will read from /api/portal/v1.3/config. Note I'm using 7.5.3 SP9. I should not initialize faulthandler policies right? But here it does. 
And then I got this exceptions with the faulthandler:

```
[ERROR] Error reading configured custom-policies. Can't parse response: {"errors":[{"code":400,"message":"Invalid policy type specified: 'faulthandler'"}]}
[ERROR] Can't initialize API-Manager API-State.
com.axway.apim.lib.AppException: Can't initialize API-Manager API-State.
    at com.axway.apim.swagger.APIManagerAdapter.getAPIManagerAPI (APIManagerAdapter.java:376)
    at com.axway.apim.actions.CreateNewAPI.execute (CreateNewAPI.java:71)
    at com.axway.apim.swagger.APIManagerAdapter.applyChanges (APIManagerAdapter.java:141)
    at com.bmw.cc.mrp.plugin.AxwayPublishingAdapter.processApi (AxwayPublishingAdapter.java:159)
    at com.bmw.cc.mrp.plugin.Publisher.publish (Publisher.java:64)
    at com.bmw.cc.mrp.plugin.Publisher.execute (Publisher.java:45)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of `com.axway.apim.swagger.api.properties.outboundprofiles.OutboundProfile`, problem: Can't initialize policies for type: faulthandler
 at [Source: (String)"{"id":"f93a3917-9da0-423d-bbab-37972cda8ee4","organizationId":"68af1395-f42f-48ed-9117-e6ad6e13ca9e","apiId":"a4566c71-0a36-4582-a1a3-b9d1b6b6ae05","name":"mrpdryrun-status-test","version":"1.0.4","vhost":null,"path":"/mrpdryrun/status/v1","descriptionType":"original","descriptionManual":null,"descriptionMarkdown":null,"descriptionUrl":null,"summary":"","retired":false,"expired":false,"image":null,"retirementDate":0,"deprecated":false,"state":"unpublished","createdOn":1564556626470,"createdBy":""[truncated 7098 chars]; line: 1, column: 1058] (through reference chain: com.axway.apim.swagger.api.state.ActualAPI["outboundProfiles"]->java.util.LinkedHashMap["_default"])
    at com.fasterxml.jackson.databind.exc.InvalidDefinitionException.from (InvalidDefinitionException.java:67)
    at com.fasterxml.jackson.databind.DeserializationContext.instantiationException (DeserializationContext.java:1608)
    at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.wrapAsJsonMappingException (StdValueInstantiator.java:484)
    at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.rewrapCtorProblem (StdValueInstantiator.java:503)
    at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.createUsingDefault (StdValueInstantiator.java:272)
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize (BeanDeserializer.java:277)
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize (BeanDeserializer.java:151)
    at com.fasterxml.jackson.databind.deser.std.MapDeserializer._readAndBindStringKeyMap (MapDeserializer.java:527)
    at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize (MapDeserializer.java:364)
    at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize (MapDeserializer.java:29)
    at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet (MethodProperty.java:129)
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize (BeanDeserializer.java:288)
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize (BeanDeserializer.java:151)
    at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose (ObjectMapper.java:4013)
    at com.fasterxml.jackson.databind.ObjectMapper.readValue (ObjectMapper.java:3004)
    at com.axway.apim.swagger.APIManagerAdapter.getAPIManagerAPI (APIManagerAdapter.java:351)
    at com.axway.apim.actions.CreateNewAPI.execute (CreateNewAPI.java:71)
    at com.axway.apim.swagger.APIManagerAdapter.applyChanges (APIManagerAdapter.java:141)
    at com.bmw.cc.mrp.plugin.AxwayPublishingAdapter.processApi (AxwayPublishingAdapter.java:159)
    at com.bmw.cc.mrp.plugin.Publisher.publish (Publisher.java:64)
    at com.bmw.cc.mrp.plugin.Publisher.execute (Publisher.java:45)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: com.axway.apim.lib.AppException: Can't initialize policies for type: faulthandler
    at com.axway.apim.swagger.api.properties.outboundprofiles.OutboundProfile.initPolicyies (OutboundProfile.java:88)
    at com.axway.apim.swagger.api.properties.outboundprofiles.OutboundProfile.<init> (OutboundProfile.java:60)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance0 (Native Method)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance (NativeConstructorAccessorImpl.java:62)
    at sun.reflect.DelegatingConstructorAccessorImpl.newInstance (DelegatingConstructorAccessorImpl.java:45)
    at java.lang.reflect.Constructor.newInstance (Constructor.java:423)
    at com.fasterxml.jackson.databind.introspect.AnnotatedConstructor.call (AnnotatedConstructor.java:119)
    at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.createUsingDefault (StdValueInstantiator.java:270)
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize (BeanDeserializer.java:277)
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize (BeanDeserializer.java:151)
    at com.fasterxml.jackson.databind.deser.std.MapDeserializer._readAndBindStringKeyMap (MapDeserializer.java:527)
    at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize (MapDeserializer.java:364)
    at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize (MapDeserializer.java:29)
    at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet (MethodProperty.java:129)
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize (BeanDeserializer.java:288)
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize (BeanDeserializer.java:151)
    at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose (ObjectMapper.java:4013)
    at com.fasterxml.jackson.databind.ObjectMapper.readValue (ObjectMapper.java:3004)
    at com.axway.apim.swagger.APIManagerAdapter.getAPIManagerAPI (APIManagerAdapter.java:351)
    at com.axway.apim.actions.CreateNewAPI.execute (CreateNewAPI.java:71)
    at com.axway.apim.swagger.APIManagerAdapter.applyChanges (APIManagerAdapter.java:141)
    at com.bmw.cc.mrp.plugin.AxwayPublishingAdapter.processApi (AxwayPublishingAdapter.java:159)
    at com.bmw.cc.mrp.plugin.Publisher.publish (Publisher.java:64)
    at com.bmw.cc.mrp.plugin.Publisher.execute (Publisher.java:45)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: com.axway.apim.lib.AppException: Can't initialize policies for type: faulthandler
    at com.axway.apim.swagger.api.properties.outboundprofiles.OutboundProfile.initPolicyies (OutboundProfile.java:85)
    at com.axway.apim.swagger.api.properties.outboundprofiles.OutboundProfile.<init> (OutboundProfile.java:60)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance0 (Native Method)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance (NativeConstructorAccessorImpl.java:62)
    at sun.reflect.DelegatingConstructorAccessorImpl.newInstance (DelegatingConstructorAccessorImpl.java:45)
    at java.lang.reflect.Constructor.newInstance (Constructor.java:423)
    at com.fasterxml.jackson.databind.introspect.AnnotatedConstructor.call (AnnotatedConstructor.java:119)
    at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.createUsingDefault (StdValueInstantiator.java:270)
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize (BeanDeserializer.java:277)
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize (BeanDeserializer.java:151)
    at com.fasterxml.jackson.databind.deser.std.MapDeserializer._readAndBindStringKeyMap (MapDeserializer.java:527)
    at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize (MapDeserializer.java:364)
    at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize (MapDeserializer.java:29)
    at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet (MethodProperty.java:129)
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize (BeanDeserializer.java:288)
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize (BeanDeserializer.java:151)
    at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose (ObjectMapper.java:4013)
    at com.fasterxml.jackson.databind.ObjectMapper.readValue (ObjectMapper.java:3004)
    at com.axway.apim.swagger.APIManagerAdapter.getAPIManagerAPI (APIManagerAdapter.java:351)
    at com.axway.apim.actions.CreateNewAPI.execute (CreateNewAPI.java:71)
    at com.axway.apim.swagger.APIManagerAdapter.applyChanges (APIManagerAdapter.java:141)
    at com.bmw.cc.mrp.plugin.AxwayPublishingAdapter.processApi (AxwayPublishingAdapter.java:159)
    at com.bmw.cc.mrp.plugin.Publisher.publish (Publisher.java:64)
    at com.bmw.cc.mrp.plugin.Publisher.execute (Publisher.java:45)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: java.lang.NullPointerException
    at com.axway.apim.swagger.api.properties.outboundprofiles.OutboundProfile.initPolicyies (OutboundProfile.java:81)
    at com.axway.apim.swagger.api.properties.outboundprofiles.OutboundProfile.<init> (OutboundProfile.java:60)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance0 (Native Method)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance (NativeConstructorAccessorImpl.java:62)
    at sun.reflect.DelegatingConstructorAccessorImpl.newInstance (DelegatingConstructorAccessorImpl.java:45)
    at java.lang.reflect.Constructor.newInstance (Constructor.java:423)
    at com.fasterxml.jackson.databind.introspect.AnnotatedConstructor.call (AnnotatedConstructor.java:119)
    at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.createUsingDefault (StdValueInstantiator.java:270)
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize (BeanDeserializer.java:277)
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize (BeanDeserializer.java:151)
    at com.fasterxml.jackson.databind.deser.std.MapDeserializer._readAndBindStringKeyMap (MapDeserializer.java:527)
    at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize (MapDeserializer.java:364)
    at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize (MapDeserializer.java:29)
    at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet (MethodProperty.java:129)
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize (BeanDeserializer.java:288)
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize (BeanDeserializer.java:151)
    at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose (ObjectMapper.java:4013)
    at com.fasterxml.jackson.databind.ObjectMapper.readValue (ObjectMapper.java:3004)
    at com.axway.apim.swagger.APIManagerAdapter.getAPIManagerAPI (APIManagerAdapter.java:351)
    at com.axway.apim.actions.CreateNewAPI.execute (CreateNewAPI.java:71)
    at com.axway.apim.swagger.APIManagerAdapter.applyChanges (APIManagerAdapter.java:141)
    at com.bmw.cc.mrp.plugin.AxwayPublishingAdapter.processApi (AxwayPublishingAdapter.java:159)
    at com.bmw.cc.mrp.plugin.Publisher.publish (Publisher.java:64)
    at com.bmw.cc.mrp.plugin.Publisher.execute (Publisher.java:45)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------

```

**Comment by cwiechmann** [comment]
Got it and will investigate. Thanks for the detailed explanation. I keep you posted here.

**Comment by cwiechmann** [comment]
@vuvdoan, I made some changes to fix the issue you described and did some tests locally. All seems to work and all integration tests are passing.  
A bit more background: 
Swagger-Promote is managing internally two sessions: One for the Admin (if present) and another one for the Org-Admin account. Based on the required action, either the Admin- or the Org-Admin-Session is used. 
Until now, Swagger-Promote has always used the Admin-Session to load the API-Manager config and ws there was no Admin-Session in your scenario, it failed.  
Now, the method: `APIManagerAdapter.getApiManagerConfig(String configField)` is deciding based on the requested configField, if an Admin-Account is required or not. If not, the Org-Admin account is used. And as you have reported, to get the version an Org-Admin account is sufficient.

[apimanager-swagger-promote-1.6.1-SNAPSHOT.tar.gz](https://github.com/Axway-API-Management-Plus/apimanager-swagger-promote/files/3468928/apimanager-swagger-promote-1.6.1-SNAPSHOT.tar.gz)

Would you be okay to test with the attached SNAPSHOT-Version and let me know?

**Comment by vuvdoan** [comment]
@cwiechmann  I ran couple of tests and it works fine now.

**Comment by cwiechmann** [comment]
This is fixed in version 1.6.1
